scipy/stats/stats.py

```diff
@@ -6099,9 +6104,13 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     if a.size == 0 or b.size == 0:
         return _ttest_nans(a, b, axis, Ttest_indResult)
 
-    if permutations:
-        if int(permutations) != permutations or permutations < 0:
-            raise ValueError("Permutations must be a positive integer.")
+    if permutations is not None:
+        if permutations < 0 or (np.isfinite(permutations) and
+                                int(permutations) != permutations) :
+            raise ValueError("Permutations must be a non-negative integer.")
+
+        if permutations == 0:
+            permutations = np.inf
```

Comment:
I added this because @jb-leger was asking for an easier way to specify that an exact test - with all permutations - was desired. So you can now do that either by setting `permutations = 0` or `permutations = np.inf`. Maybe we only need one of those (probably `np.inf` would make more sense)?

Reply:
OK, next commit will do the exact test for `permutations=np.inf` and use t-distribution for `permutations=0`.